### ISSUE TITLE
L1 loss function helps quick alignment

### ISSUE THREAD

**Issue by begeekmyfriend** [Rayhane-mamah/Tacotron-2#336]
As L2 loss very sensitive to outliers, I choosed L1 loss function for mel spectrograms for my imperfect dataset and it learned alignment quickly. So if it gets hard to alignment, please change your loss function.
```python
before = tf.reduce_mean(tf.abs(self.mel_targets - self.decoder_output))
after = tf.reduce_mean(tf.abs(self.mel_targets - self.mel_outputs))
```
![step-9000-align](https://user-images.githubusercontent.com/6031938/53158524-c84d8680-35fe-11e9-97fd-5dad2edef73d.png)
```python
before = tf.losses.mean_squared_error(self.mel_targets, self.decoder_output)
after = tf.losses.mean_squared_error(self.mel_targets, self.mel_outputs)
```
![step-48000-align](https://user-images.githubusercontent.com/6031938/53158698-342fef00-35ff-11e9-8a9c-c77d07b4ee8a.png)

**Comment by begeekmyfriend** [comment]
Quoted from this [blog](http://rishy.github.io/ml/2015/07/28/l1-vs-l2-loss/)

> With outliers in the dataset, a L2(Loss function) tries to adjust the model according to these outliers on the expense of other good-samples, since the squared-error is going to be huge for these outliers(for error > 1). On the other hand L1(Least absolute deviation) is quite resistant to outliers.
> 
> As a result, L2 loss function may result in huge deviations in some of the samples which results in reduced accuracy.
> 
> So, if you can ignore the ouliers in your dataset or you need them to be there, then you should be using a L1 loss function, on the other hand if you don’t want undesired outliers in the dataset and would like to use a stable solution then first of all you should try to remove the outliers and then use a L2 loss function. Or performance of a model with a L2 loss function may deteriorate badly due to the presence of outliers in the dataset.

**Comment by begeekmyfriend** [comment]
Another dataset, just within 6K steps!
![step-6000-align](https://user-images.githubusercontent.com/6031938/53222201-afe57680-36a7-11e9-9520-afb51bbb7c97.png)

**Comment by begeekmyfriend** [comment]
Further testing shows that MAE would lead to vibration on loss curve while MSE not. So the solution for quick alignment should be guided attention loss plus MSE. Close this issue.
![Screenshot from 2019-03-30 19-08-59](https://user-images.githubusercontent.com/6031938/55275316-624fcf80-531f-11e9-80ec-eba8a1e595c3.png)
![Screenshot from 2019-03-30 19-10-16](https://user-images.githubusercontent.com/6031938/55275323-78f62680-531f-11e9-9025-00ffd760b1f7.png)
